### PR TITLE
Fix query_id from PG to TServer and add flush and compaction wait event class

### PR DESCRIPTION
**Pull request by abhinab-yb**
pg_stat_statements doesn't store the query id in the `Query` object in case of utility statements. This PR fixes that and also setting the query_id correctly in AUH metadata.

A new wait event class is also introduced for the background rocksdb flush and compaction workers.

### CODE REVIEW

File: src/postgres/src/backend/parser/analyze.c [hunk 1]
```diff
@@ -156,12 +159,6 @@ parse_analyze_varparams(RawStmt *parseTree, const char *sourceText,
 
 	query = transformTopLevelStmt(pstate, parseTree);
 
-	if (IsYugaByteEnabled())
```
Comment (vrajat) [inline src/postgres/src/backend/parser/analyze.c:159]: Why was this code deleted? If you have moved it somewhere else, can you also set queryid in `MyProc` ?

File: src/yb/yql/pggate/pg_session.cc [hunk 2]
```diff
@@ -668,7 +668,7 @@ Result<PerformFuture> PgSession::Perform(BufferableOperations&& ops, PerformOpti
       caching_info.mutable_lifetime_threshold_ms()->set_value(*cache_options.lifetime_threshold_ms);
     }
   }
-
+
```
Comment (vrajat) [inline src/yb/yql/pggate/pg_session.cc:671]: nit: Whitespace change

**Review by vrajat — CHANGES_REQUESTED** [review]
Can you add back `MyProc->queryid = ...` and also add it to the new places where queryid is set.

**Review by amitanandaiyer — APPROVED** [review]
lgtm

**Pull request merged**